### PR TITLE
Fix unstoppable empty footnote loop (#5354)

### PR DESCRIPTION
**Pull request by wznmickey**
Fixes https://github.com/typst/typst/issues/5354

Follow https://github.com/typst/typst/issues/5354#issuecomment-2453364239, a `exist_non_empty_frame` is used to check if there are any non-empty frames. It is checked before `first` to make sure do not have reference conflicts with this iterator.

### CODE REVIEW

File: crates/typst-layout/src/flow/compose.rs [hunk 1]
```diff
@@ -431,6 +431,15 @@ impl<'a, 'b> Composer<'a, 'b, '_, '_> {
...
         let nested = find_in_frames::<FootnoteElem>(&frames);
 
+        // Check if there are any non-empty frames.
+        let mut exist_non_empty_frame = false;
+        for i in &frames {
+            if !i.is_empty() {
+                exist_non_empty_frame = true;
+                break;
+            }
+        }
```
Comment (Coekjan) [inline crates/typst-layout/src/flow/compose.rs:441]:
```suggestion
        let exist_non_empty_frame = frames.iter().any(|i| !i.is_empty());
```

Comment (wznmickey) [reply crates/typst-layout/src/flow/compose.rs]: Changed

**Comment by laurmaedje** [comment]
It would be great if you could also add a regression test in `tests/suite/layout/flow/footnote.typ` for this.

**Comment by wznmickey** [comment]
@laurmaedje Added.

**Comment by laurmaedje** [comment]
Thanks!

**Pull request merged**